sovrin_client/agent/walleted.py

```diff
@@ -923,30 +928,31 @@ def _(reply, err):
         return _
 
     def _updateLinkWithLatestInfo(self, link: Link, reply):
-        link.remoteVerkey = DidVerifier(reply[VERKEY],
-                                        identifier=link.remoteIdentifier).verkey
         if DATA in reply and reply[DATA]:
-            data = json.loads(reply[DATA])
-            ep = data.get(ENDPOINT)
-            if ep:
-                # TODO: Validate its an IP port pair or a malicious entity
-                # can crash the code
-                if 'ha' in ep:
-                    ip, port = ep['ha'].split(":")
-                    link.remoteEndPoint = (ip, int(port))
-                if PUBKEY in ep:
-                    link.remotePubkey = ep[PUBKEY]
-                else:
-                    link.remotePubkey = friendlyVerkeyToPubkey(
-                        link.remoteVerkey) if link.remoteVerkey else None
-
-        link.linkLastSynced = datetime.now()
-        self.notifyMsgListener("    Link {} synced".format(link.name))
-        # TODO need to move this to after acceptance,
-        # unless we want to support an anonymous ping
-        # if link.remoteEndPoint:
-        #     reqId = self._pingToEndpoint(link.name, link.remoteEndPoint)
-        #     return reqId
+            try:
+                data = json.loads(reply[DATA])
+
+                verkey = data.get(VERKEY)
+                if verkey is not None:
+                    link.remoteVerkey = data[VERKEY]
+
+                ep = data.get(ENDPOINT)
+                if ep is not None:
+                    # TODO: Validate its an IP port pair or a malicious entity
+                    # can crash the code
+                    if 'ha' in ep:
+                        ip, port = ep['ha'].split(":")
+                        link.remoteEndPoint = (ip, int(port))
+                    if PUBKEY in ep:
+                        link.remotePubkey = ep[PUBKEY]
+                    else:
+                        link.remotePubkey = friendlyVerkeyToPubkey(
+                            link.remoteVerkey) if link.remoteVerkey else None
+
+                link.linkLastSynced = datetime.now()
+                self.notifyMsgListener("    Link {} synced".format(link.name))
+            except ValueError:
+                pass
```

Comment:
Too broad of a code block to check for exception. Also need a log message here to help debugging